--- ../ext/util/make_ext	2010-04-04 01:40:10.132379299 +0200
+++ make_ext.android	2010-04-04 01:39:43.625630126 +0200
@@ -32,6 +32,9 @@
 	;;
 esac
 
+ARCHI=android-3
+CFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DPAGE_SIZE=0x400 -I$HOME/android-ndk/build/platforms/$ARCHI/arch-arm/usr/include -mthumb -Os -fomit-frame-pointer -fno-strict-aliasing -finline-limit=64 -nostdlib -fpic -mthumb-interwork -ffunction-sections -funwind-tables -fstack-protector -fno-short-enums -D__ARM_ARCH_5__ -D__ARM_ARCH_5T__ -D__ARM_ARCH_5E__ -D__ARM_ARCH_5TE_ -march=armv5te -mtune=xscale -msoft-float"
+LDFLAGS="-nostdlib -Wl,-dynamic-linker,/system/bin/linker -L$HOME/android-ndk/build/platforms/$ARCHI/arch-arm/usr/lib -lc -lgcc -lm -ldl"
 
 case $CONFIG in
 '')
@@ -96,7 +99,7 @@
 # 'dynamic', 'static', and 'static_pic' (the last one respects
 # CCCDLFLAGS such as -fPIC -- see static_target in the main Makefile.SH)
 case "$target" in
-dynamic)    makeargs="LINKTYPE=dynamic";
+dynamic)    makeargs="LINKTYPE=dynamic PERL_ARCHIVE=$HOME/android-ndk/apps/perldroid/jni/libperl/perl-5.10.0/libperl.so";
 	    target=all
 	    ;;
 static)     makeargs="LINKTYPE=static CCCDLFLAGS="
@@ -126,7 +129,8 @@
 esac
 
 if test ! -f $makefile ; then
-	test -f Makefile.PL && $run ../$depth/miniperl -I../$depth/lib Makefile.PL INSTALLDIRS=perl INSTALLMAN3DIR=none PERL_CORE=1 $passthru
+#	test -f Makefile.PL && $run ../$depth/miniperl -I../$depth/lib Makefile.PL INSTALLDIRS=perl INSTALLMAN3DIR=none PERL_CORE=1 $passthru
+	test -f Makefile.PL && $run ../$depth/miniperl -I../$depth/fake_config_library -MConfig Makefile.PL INSTALLDIRS=perl INSTALLMAN3DIR=none PERL_CORE=1 $passthru CC="$HOME/android-ndk/build/prebuilt/linux-x86/arm-eabi-4.4.0/bin/arm-eabi-gcc $CFLAGS" LD="$HOME/android-ndk/build/prebuilt/linux-x86/arm-eabi-4.4.0/bin/arm-eabi-gcc $LDFLAGS" PERLRUN="/usr/bin/perl -I../$depth/fake_config_library -MConfig" FULLPERLRUN="/usr/bin/perl -I../$depth/fake_config_library -MConfig" NOECHO= MKPATH="mkdir -p" PERL="/usr/bin/perl -I/usr/lib/perl/5.10.0" ABSPERLRUN="/usr/bin/perl -I/usr/lib/perl/5.10.0"
 fi
 if test ! -f $makefile ; then
 	echo "Warning: No Makefile!"
