--- ../installperl	2012-04-25 02:18:34.000000000 +0200
+++ installperl.android	2012-06-21 17:47:07.394725824 +0200
@@ -3,8 +3,8 @@
 BEGIN {
     require 5.004;
     chdir '..' if !-d 'lib' and -d '../lib';
-    @INC = 'lib';
-    $ENV{PERL5LIB} = 'lib';
+#    @INC = 'lib';
+#    $ENV{PERL5LIB} = 'lib';
 
     # This needs to be at BEGIN time, before any use of Config
     require './install_lib.pl';
@@ -868,7 +868,7 @@
 		print " " if (@opts);
 		print "$file\n";
 	    }
-	    system("strip", @opts, $file);
+	    system("arm-linux-androideabi-strip", @opts, $file);
 	} else {
 	    print "# file '$file' skipped\n" if $opts{verbose};
 	}
